src/redteam/plugins/base.ts

```diff
@@ -134,6 +134,44 @@ export function parseGeneratedPrompts(generatedPrompts: string): { prompt: strin
     .map((prompt) => ({ prompt }));
 }
 
+/**
+ * Parses LLM output into multi-input test cases when inputs schema is defined.
+ * Extracts JSON from <Prompt> tags and returns them as prompt strings.
+ *
+ * @param generatedOutput - The LLM response containing generated test cases.
+ * @param inputs - The inputs schema defining expected variable names.
+ * @returns An array of { prompt: string } objects where prompt is the JSON string.
+ */
+export function parseGeneratedInputs(
+  generatedOutput: string,
+  inputs: Record<string, string>,
+): { prompt: string }[] {
+  const results: { prompt: string }[] = [];
+  const inputKeys = Object.keys(inputs);
+
+  // Extract JSON from <Prompt> tags
+  const promptRegex = /<Prompt>([\s\S]*?)<\/Prompt>/gi;
```

Comment:
This extraction is also contained within `processPromptForInputs`; would be nice as a DRY function w/ test cov.